openedx/features/content_type_gating/partitions.py

```diff
@@ -29,6 +33,9 @@ def create_content_gating_partition(course):
     Create and return the Content Gating user partition.
     """
 
+    if not (CONTENT_TYPE_GATING_FLAG.is_enabled() or CONTENT_TYPE_GATING_STUDIO_UI_FLAG.is_enabled()):
```

Comment:
why not only check the STUDIO_UI flag here?  
you'd have the flexibility to turn this on or off independently then

Reply:
Oops. I meant to only exclude the partition if they are both disabled. Booleans are hard...

Reply:
Oh, no, I got it right. We only disable the partition if both flags are disabled.

Reply:
Sorry, to be clear: This code only affects whether the Partition exists as an entity in the system. Separately, the `STUDIO_UI` flag controls whether the partition is visible in Studio. If we turn the whole system on, then we need the partition to exist in order to power the LMS code that uses it.

Reply:
that clarifies things, thanks!